### PR TITLE
bibliophile.getPositionSizes

### PR DESCRIPTION
**Pull request by atvanguard**
## Why this should be merged
- Implements `bibliophile.getPositionSizes`
- Reduces fixed gas cost of `GetNotionalPositionAndMarginGasCost` and `GetPositionSizesGasCost` to 1000
- Change genesis addresses to start from `0x0300000000000000000000000000000000000000`

## How this was tested
Running `hubble-protocol/scripts/benchmarks/bibliophile.js`. Currently, the gas estimates are as follows:
* `orderPlaced` ~188k gas
* `orderMatched` ~594-612k gas

### CODE REVIEW

File: plugin/evm/orderbook_test.go [hunk 1]
```diff
@@ -796,7 +796,7 @@ var (
 		  }
 		]
 	  }`
-	orderBookAddress common.Address = common.HexToAddress("0x0300000000000000000000000000000000000069")
+	orderBookAddress common.Address = common.HexToAddress("0x0300000000000000000000000000000000000000")
```
Comment (shubhamgoyal42) [inline plugin/evm/orderbook_test.go:799]: also need to change this at line 807 in the genesis json string

**Review by fdgod09 — APPROVED** [review]

File: precompile/contracts/hubblebibliophile/margin_account.go [hunk 2]
```diff
@@ -10,8 +10,8 @@ import (
 )
 
 const (
-	MARGIN_ACCOUNT_GENESIS_ADDRESS        = "0x0300000000000000000000000000000000000070"
-	VAR_MARGIN_MAPPING_STORAGE_SLOT int64 = 11
+	MARGIN_ACCOUNT_GENESIS_ADDRESS        = "0x0300000000000000000000000000000000000001"
+	VAR_MARGIN_MAPPING_STORAGE_SLOT int64 = 10
```
Comment (fdgod09) [inline precompile/contracts/hubblebibliophile/margin_account.go:14]: Did this slot change on contract side from 11 to 10 ?
could find the change in PR related to this in hubble-protocol. Is this a fix ?

File: precompile/contracts/hubblebibliophile/clearing_house.go [hunk 3]
```diff
@@ -95,3 +94,11 @@ func GetMaintenanceMargin(stateDB contract.StateDB) *big.Int {
 func GetMinAllowableMargin(stateDB contract.StateDB) *big.Int {
 	return new(big.Int).SetBytes(stateDB.GetState(common.HexToAddress(CLEARING_HOUSE_GENESIS_ADDRESS), common.BytesToHash(common.LeftPadBytes(big.NewInt(MIN_ALLOWABLE_MARGIN_SLOT).Bytes(), 32))).Bytes())
 }
+
+func GetPositionSizes(stateDB contract.StateDB, trader *common.Address) []*big.Int {
```
Comment (fdgod09) [inline precompile/contracts/hubblebibliophile/clearing_house.go:98]: if we are not using this in subnet-evm side we can make this private function.

File: precompile/contracts/hubblebibliophile/margin_account.go (hunk 2 above)
Comment (atvanguard) [reply precompile/contracts/hubblebibliophile/margin_account.go:14]: It is a fix. Somehow got crept in here https://github.com/hubble-exchange/subnet-evm/pull/52/files#diff-558ab4d135487100d7a0aa2d7dc2794369385b2cad9a346ad53cac299d94995eL14

File: plugin/evm/orderbook_test.go (hunk 1 above)
Comment (atvanguard) [reply plugin/evm/orderbook_test.go:799]: done

File: precompile/contracts/hubblebibliophile/clearing_house.go (hunk 3 above)
Comment (atvanguard) [reply precompile/contracts/hubblebibliophile/clearing_house.go]: done

**Pull request merged**